src/main/scala/org/constellation/snapshotstreaming/FileBasedLastGlobalSnapshotStorage.scala

```diff
@@ -64,11 +66,12 @@ object FileBasedLastGlobalSnapshotStorage {
         Files[F]
           .readAll(path)
           .through(text.utf8.decode)
-          .map(decode[Hashed[GlobalSnapshot]])
-          .map(_.toOption)
           .compile
-          .last
-          .map(_.flatten)
+          .toList
+          .map(_.mkString)
+          .map(decode[Hashed[GlobalSnapshot]])
+          .flatMap(_.liftTo[F])
```

Comment:
isn't `.flatMap(_.liftTo[F])` equal to `.map(a => a)` which is just an identity?